device/peripherals/ADXL375/ADXL375.h

```diff
@@ -0,0 +1,163 @@
+//
+// Created by skippynose on 12/17/22.
+//
+
+#ifndef LAUNCH_CORE_ADXL375_H
+#define LAUNCH_CORE_ADXL375_H
+
+
+#include <stdlib.h>
+#include <stdint.h>
+
+#include "sched/macros.h"
+#include "return.h"
+#include "device/I2CDevice.h"
+
+
+class ADXL375 {
+public:
+    ADXL375(I2CDevice& i2c) : m_i2c(i2c)
+        {}
+
+   RetType readX(int16_t* xAxis){
+        // TODO read the values of the MSB and LSB and pass it into the final variable
+        RetType ret;
+        RESUME();
+        uint8_t lsb;
+        uint8_t msb;
+        uint16_t value;
+
+        // reading the data
+        ret = CALL(m_i2c.read(xLSBData, lsb, 1));
+        if (ret != RET_SUCCESS){
+            RESET();
+            return ret;
+        }
+
+        ret = CALL(m_i2c.read(xMSBData, msb, 1));
+        if (ret != RET_SUCCESS){
+            RESET();
+            return ret;
+        }
+
+        // combining the data
+        value |= msb << 8;
+        value |= lsb;
+
+        // value is in 2's complement so have to convert it
+        int16_t adjustedValue = value;
+        adjustedValue *= -1;
+        *xAxis = adjustedValue;
+
+        RESET();
+       return RET_SUCESS;
+    }
+
+    RetType readY(int16_t* yAxis){
+        // TODO read the values of the MSB and LSB and pass it into the final variable
+        RetType ret;
+        RESUME();
+        uint8_t lsb;
+        uint8_t msb;
+        uint16_t value;
+
+        // reading the data
+        ret = CALL(m_i2c.read(yLSBData, lsb, 1));
+        if (ret != RET_SUCCESS){
+            RESET();
+            return ret;
+        }
+
+        ret = CALL(m_i2c.read(yMSBData, msb, 1));
+        if (ret != RET_SUCCESS){
+            RESET();
+            return ret;
+        }
+
+        // combining the data
+        value |= msb << 8;
+        value |= lsb;
+
+        // value is in 2's complement so have to convert it
+        int16_t adjustedValue = value;
+        adjustedValue *= -1;
+        *yAxis = adjustedValue;
+
+        RESET();
+        return RET_SUCCESS;
+    }
+
+    RetType readZ(int16_t* zAxis){
+        // TODO read the values of the MSB and LSB and pass it into the final variable
+        RetType ret;
+        RESUME();
+        uint8_t lsb;
+        uint8_t msb;
+        uint16_t value;
+
+        // reading the data
+        ret = CALL(m_i2c.read(zLSBData, lsb, 1));
+        if (ret != RET_SUCCESS){
+            RESET();
+            return ret;
+        }
+
+        ret = CALL(m_i2c.read(zMSBData, msb, 1));
+        if (ret != RET_SUCCESS){
+            RESET();
+            return ret;
+        }
+
+        // combining the data
+        value |= msb << 8;
+        value |= lsb;
+
+        // value is in 2's complement so have to convert it
+        int16_t adjustedValue = value;
```

Comment:
Whats the point of having a value variable? Also your value variable is uninitialized in the first place, so it might lead to unintended behavior